app/helpers.py

```diff
@@ -0,0 +1,68 @@
+from functools import wraps
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, make_response
+
+# Decorator to check if task exists.
+def require_valid_task_id(endpoint):
+    @wraps(endpoint)
+    def fn(*args, task_id, **kwargs):
+
+        task = Task.query.get(task_id)
+
+        if not task: 
+            return make_response("", 404)
+
+        return endpoint(*args, task=task, **kwargs)
+    return fn
+
+# Decorator to check if goal exists.
+def require_valid_goal_id(endpoint):
+    @wraps(endpoint)
+    def fn(*args, goal_id, **kwargs):
+
+        goal = Goal.query.get(goal_id)
+
+        if not goal: 
+            return make_response("", 404)
+
+        return endpoint(*args, goal=goal, **kwargs)
+    return fn
+
+# Decorator to check if the request body to create a new goal includes "title".
+def require_valid_goal_request_body(endpoint):
+    @wraps(endpoint)
+    def fn(*args, **kwargs):
+        request_body = request.get_json()
+
+        if "title" not in request_body:
+            return {"details": "Invalid data"}, 400
+
+        return endpoint(*args, request_body=request_body, **kwargs)
+
+    return fn
+
+
+# Create response body for details in error message responses.
+def create_details_response(details):
```

Comment:
Even small helpers like this can really add to our quality of life. Each time we use this function, that's one fewer chance to accidentally have a typo in the dictionary keys.